pkg/core/annotation.go

```diff
@@ -17,6 +17,9 @@ type (
 		Capability *annotation.Capability
 		// Node is the node that has been annotated; not the comment node representing the annotation itself.
 		Node *sitter.Node
+		// IsTransformed indicates whether the content of an annotation's original Node field been transformed.
+		// When reparsing a transformed annotation, its Node field will be nil to prevent plugins from treating the transformed code as input.
+		IsTransformed bool
```

Comment:
What do we need this for? The node file gets updated when the code is transformed so it's always correct.

Unless you mean `IsDetached` - not all transformations would result in detached nodes.